src/SFA.DAS.QnA.Api.UnitTests/ApplicationDataControllerTests/WhenPostingApplicationData.cs

```diff
@@ -0,0 +1,78 @@
+﻿using FluentAssertions;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using NSubstitute;
+using NUnit.Framework;
+using SFA.DAS.QnA.Api.Controllers;
+using SFA.DAS.QnA.Api.Types;
+using SFA.DAS.QnA.Application.Commands.SetApplicationData;
+using System;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace SFA.DAS.QnA.Api.UnitTests.ApplicationDataControllerTests
+{
+    public class WhenPostingApplicationData
+    {
+        [Test]
+        public async Task And_MediatorCallFails_ThenNotFoundResultReturned()
+        {
+            var mediator = Substitute.For<IMediator>();
+            var controller = new ApplicationDataController(mediator);
+            var applicationData = File.ReadAllText("ApplicationDataControllerTests/test.json");
```

Comment:
Since the result of this test doesn't depend on the exact contents of this data, it would be more accurate to avoid passing this entire file. I would pass something like `@"{""anyName"": ""does not matter""}` to the Post method, to clearly indicate that